test/heat_equation_analytical.jl

```diff
@@ -0,0 +1,168 @@
+using Ferrite, Test
+
+module ConvergenceTestHelper
+
+using Ferrite, SparseArrays, ForwardDiff, Test
+import LinearAlgebra: diag
+
+get_geometry(::Ferrite.Interpolation{RefLine}) = Line
+get_geometry(::Ferrite.Interpolation{RefQuadrilateral}) = Quadrilateral
+get_geometry(::Ferrite.Interpolation{RefTriangle}) = Triangle
+get_geometry(::Ferrite.Interpolation{RefPrism}) = Wedge
+get_geometry(::Ferrite.Interpolation{RefHexahedron}) = Hexahedron
+get_geometry(::Ferrite.Interpolation{RefTetrahedron}) = Tetrahedron
+
+get_quadrature_order(::Lagrange{shape, order}) where {shape, order} = 2*order
+get_quadrature_order(::Serendipity{shape, order}) where {shape, order} = 2*order
+get_quadrature_order(::CrouzeixRaviart{shape, order}) where {shape, order} = 2*order+1
+
+get_N(::Ferrite.Interpolation{shape, 1}) where {shape} = 19
+get_N(::Ferrite.Interpolation{shape, 2}) where {shape} = 12
+get_N(::Ferrite.Interpolation{shape, 3}) where {shape} = 8
+get_N(::Ferrite.Interpolation{shape, 4}) where {shape} = 5
+get_N(::Ferrite.Interpolation{shape, 5}) where {shape} = 3
+
+analytical_solution(x) = prod(cos, x*π/2)
+analytical_rhs(x) = -sum(diag(ForwardDiff.hessian(analytical_solution,x)))
+
+# Standard assembly copy pasta for Poisson problem
+function assemble_element!(Ke::Matrix, fe::Vector, cellvalues::CellValues, coords)
+    n_basefuncs = getnbasefunctions(cellvalues)
+    ## Reset to 0
+    fill!(Ke, 0)
+    fill!(fe, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the quadrature weight
+        dΩ = getdetJdV(cellvalues, q_point)
+        x = spatial_coordinate(cellvalues, q_point, coords)
+        ## Loop over test shape functions
+        for i in 1:n_basefuncs
+            δu  = shape_value(cellvalues, q_point, i)
+            ∇δu = shape_gradient(cellvalues, q_point, i)
+            ## Add contribution to fe
+            fe[i] += analytical_rhs(x) * δu * dΩ
```

Comment:
Looks like `forwardDiff` doesn't play work that well with `Vec`
```julia
julia> @benchmark ConvergenceTestHelper.analytical_rhs(t) setup=(t = Vec(1.0,2.0) ;)
BenchmarkTools.Trial: 10000 samples with 15 evaluations.
 Range (min … max):  973.067 ns … 236.959 μs  ┊ GC (min … max): 0.00% … 98.26%
 Time  (median):       1.036 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.196 μs ±   5.064 μs  ┊ GC (mean ± σ):  9.35% ±  2.20%

   ▅█                                                            
  ▄███▄▃▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▁
  973 ns           Histogram: frequency by time         2.53 μs <

 Memory estimate: 1.78 KiB, allocs estimate: 20.

julia> using StaticArrays

julia> @benchmark ConvergenceTestHelper.analytical_rhs(t) setup=(t = SVector{2}(Vec(1.0,2.0)) ;)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  1.266 ns … 30.051 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     1.272 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.293 ns ±  0.586 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

   ▃█▇█▆▆▂▃▂▃▁                   ▁                           ▂
  ▆████████████▄▃▁▃▁▁▁▁▁▃▆█▇█▇▇█▇█▇▇▁▃▁▁▁▁▃▁▁▄▁▄▁▇▆▇▅▆▄▅▄▆▁▆ █
  1.27 ns      Histogram: log(frequency) by time     1.35 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
 ```
 
 Timing the assembly for `Lagrange{RefTriangle, 5}()`, N = 20:
 ```julia
   2.177157 seconds (32.24 M allocations: 2.280 GiB, 11.30% gc time, 25.05% compilation time)
```
wrapping `x` in `SVector`:
```julia
  0.097493 seconds (841 allocations: 18.978 MiB)
```

Reply:
Interesting! Defining `analytical_rhs(x) = -tr(hessian(analytical_solution,x))` is another way to solve that, which uses `Tensors`'s `hessian` function (in the end probably compiles to the same code as SVector). Otherwise, regular `Array`s are created as intermediate results.

Reply:
Yea, I should just use laplace here directly.